packages/file-storage-common/src/services/attachment-service.ts

```diff
@@ -0,0 +1,159 @@
+import { EncryptionKey } from '@meeco/cryppo';
+import {
+  AttachmentDirectDownloadUrl,
+  AttachmentDirectUploadUrl,
+  DirectAttachment,
+  DirectAttachmentsApi,
+} from '@meeco/vault-api-sdk';
+import { buildApiConfig, IFileStorageAuthConfiguration } from '../auth';
+import { AzureBlockUpload } from '../azure-block-upload';
+
+interface IDirectAttachmentUploadData {
+  directUploadUrl: string;
+  file: File | string;
+  encrypt: boolean;
+  attachmentDek?: EncryptionKey;
+}
+
+interface IDirectAttachmentUploadResponse {
+  artifacts: {
+    iv: Buffer[];
```

Comment:
Ideally, we shouldn't use or refer to `Buffer` at all in common, since it doesn't exist in the browser. Also no real point in webpack trying to magically include its own polyfill for it when it is a subtype of UInt8Array anyway.

Reply:
Good spot! Amusingly it was actually being illegally cast from a Uint8Array to a Buffer by this interface :D